xarray/core/computation.py

```diff
@@ -579,28 +592,55 @@ def apply_variable_ufunc(
                 "``.load()`` or ``.compute()``"
             )
         elif dask == "parallelized":
-            input_dims = [broadcast_dims + dims for dims in signature.input_core_dims]
             numpy_func = func
 
+            if dask_gufunc_kwargs is None:
+                dask_gufunc_kwargs = {}
+
+            output_sizes = dask_gufunc_kwargs.pop("output_sizes", None)
+            if output_sizes is not None:
+                output_sizes = {
+                    "dim{}".format(signature.dims_map[key]): value
```

Comment:
```suggestion
                    signature.dims_map.get(key, key): value
```

If there is a `output_sizes` that is missing from the core_dims this will raise a informative error in dask (instead of a non-informative error here)